Framework/PythonInterface/test/python/plugins/algorithms/WorkflowAlgorithms/D7AbsoluteCrossSectionsTest.py

```diff
@@ -0,0 +1,105 @@
+                                     CrossSectionSeparationMethod='XYZ', NormalisationMethod='None')
+        self.assertTrue(mtd['xyz'].getNumberOfEntries() == 3)
+        self._check_output('xyz', 1, 132, onlySeparation=True)
+
+    def test_10p_separation(self):
+        D7AbsoluteCrossSections(InputWorkspace='vanadium_10p', OutputWorkspace='10p',
+                                     CrossSectionSeparationMethod='10p', ThetaOffset=1.0,
+                                     NormalisationMethod='None')
+        self.assertTrue(mtd['10p'].getNumberOfEntries() == 3)
+        self._check_output('10p', 1, 132, onlySeparation=True)
+
+    def test_10p_separation_double_xyz(self):
+        D7AbsoluteCrossSections(InputWorkspace='vanadium_xyz', RotatedXYZWorkspace='vanadium_xyz',
+                                     OutputWorkspace='10p_double_xyz',
+                                     CrossSectionSeparationMethod='10p', ThetaOffset=1.0,
+                                     NormalisationMethod='None')
+        self.assertTrue(mtd['10p_double_xyz'].getNumberOfEntries() == 3)
+        self._check_output('10p_double_xyz', 1, 132, onlySeparation=True)
+
+    def test_vanadium_normalisation(self):
+        D7AbsoluteCrossSections(InputWorkspace='sample_data', OutputWorkspace='normalised_sample_vanadium',
+                                     CrossSectionSeparationMethod='XYZ', NormalisationMethod='Vanadium',
+                                     VanadiumInputWorkspace='vanadium_data', AbsoluteUnitsNormalisation=False)
+        self.assertTrue(mtd['normalised_sample_vanadium'].getNumberOfEntries() == 6)
+        self._check_output('normalised_sample_vanadium', 1, 132, onlySeparation=False)
+
+    def test_paramagnetic_normalisation(self):
+        D7AbsoluteCrossSections(InputWorkspace='sample_data', OutputWorkspace='normalised_sample_magnetic',
+                                     CrossSectionSeparationMethod='XYZ', NormalisationMethod='Paramagnetic',
+                                     SampleSpin = 0.5, AbsoluteUnitsNormalisation=False)
+        self.assertTrue(mtd['normalised_sample_magnetic'].getNumberOfEntries() == 6)
+        self._check_output('normalised_sample_magnetic', 1, 132, onlySeparation=False)
+
+    def test_incoherent_normalisation(self):
+        D7AbsoluteCrossSections(InputWorkspace='sample_data', OutputWorkspace='normalised_sample_incoherent',
+                                     CrossSectionSeparationMethod='XYZ', NormalisationMethod='Incoherent',
+                                     AbsoluteUnitsNormalisation=False)
+        self.assertTrue(mtd['normalised_sample_incoherent'].getNumberOfEntries() == 6)
+        self._check_output('normalised_sample_incoherent', 1, 132, onlySeparation=False)
+
+    def _check_output(self, ws, blocksize, spectra, onlySeparation):
+        self.assertTrue(mtd[ws])
+        self.assertTrue(isinstance(mtd[ws], WorkspaceGroup))
```

Comment:
Could you assert also on number of workspaces in the group?

Reply:
This is checked before the _check_output method is invoked, but can be made a part it for improved clarity.